### DISCUSSION TITLE
Would like to work on Pluto, Any scheme of how the flow between browser and Julia works?

### DISCUSSION THREAD

**Discussion by zahachtah** [fonsp/Pluto.jl#1307] [Pluto development] [answered]
Is there some schematics on how the states are defined in Julia and the browser and how the message-definitions look when sending information between them?

If not I'll try to dig through the code but I suspect someone must have made some diagram or table or something defining all the specs?

**Comment by lungben** [comment, marked as answer]
I am not aware of such a diagram, but agree that this would be helpful.

For the Julia side, the conversion to JS data structures is done here: https://github.com/fonsp/Pluto.jl/blob/28e452bc0582bb53c25768e02169a65ffa436e99/src/webserver/Dynamic.jl#L95
For the frontend side, this may be a starting point: https://github.com/fonsp/Pluto.jl/blob/main/frontend/components/Editor.js

**Reply by zahachtah** [reply]
Thanks m8, that's very helpful pointer where to start!
